heap_sort/HeapSort.java

```diff
@@ -0,0 +1,44 @@
+/*
+*  Following is the implementation of HeapSort algorithm
+*  It will sort the input array with time complexity O(N*logN)
+*  N being the size of array
+*/
+
+public class HeapSort {
+
+	public static void heapify(int[] arr, int size, int parent) {  // Build max-heap of subtree of parent index
```

Comment:
Using max_heap will help a reader understanding the code more easily.

> if we want to sort in descending we would need min heap which can be done from this one only by few changes.

I think those few changes can contain changing `max_heap()` to `min_heap()`.